MovieApp/MovieApp/App/Features/Movies/UseCases/MoviesUseCase.swift

```diff
@@ -1,13 +1,33 @@
 import UIKit
+import Combine
 
 class MoviesUseCase: MoviesUseCaseProtocol {
     
     private let moviesRepository: MoviesRepositoryProtocol
     private let userDefaultsRepository: UserDefaultsRepositoryProtocol
     
-    var favoriteItems: [Int] {
+    var favoriteMovies: AnyPublisher<[FavoriteMovieModel], Never> {
         userDefaultsRepository
             .favoriteItems
+            .setFailureType(to: Error.self)
+            .flatMap { [weak self] ids -> AnyPublisher<[MovieDetailsRepositoryModel], Error> in
+                guard let self = self else { return .never() }
+                
+                let movieStreams = ids.map {
+                    self.moviesRepository.fetchfavoriteMovie(with: $0)
+                }
+                return Publishers.MergeMany(movieStreams)
+                    .collect()
+                    .eraseToAnyPublisher()
+            }
+            .assertNoFailure()
```

Comment:
We do not use this operator in production. If you do not want to propagate error, use .ignoreErrors().